### PR TITLE
Update equality check for person

### LINKED ISSUE #258: [PE-D][Tester B] Multiple people allowed with same name
Entire Address Book:

Steps to Reproduce:
`add n/John Doe p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 r/1 s/M c/2A`
`add n/John Doe p/98765431 e/johnd@example.com a/311, Clementi Ave 2, #02-25 r/1 s/M c/2A`

Expected: person should still be added even though they have the same name, as it is possible for 2 people to have the same exact same name. 
Actual: person with same name is not allowed; error message thrown
Suggestion: to allow same name, but if they have the same phone/email, then they should not be accepted since it is unlikely for 2 people to have the same name and phone/email number.

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: Sornsornah/ped#8

### PR DESCRIPTION
**Pull request by Shruthiiii03**
2 students are now considered the same person only if they are in the same class and have the same register number. 
Fixes #258

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -190,16 +190,20 @@ public HashMap<AbsentDate, AbsentReason> getAttendances() {
...
         }
 
-        return otherPerson != null
-                && otherPerson.getName().equals(getName());
+        boolean sameClass = (otherPerson.getStudentClass().equals(this.getStudentClass()));
+        boolean sameRegNo = (otherPerson.getRegisterNumber().equals(this.getRegisterNumber()));
+        boolean samePhone = (otherPerson.getPhone().equals(this.getPhone()));
+        boolean sameEmail = (otherPerson.getEmail().equals(this.getEmail()));
+
+        return sameClass && sameRegNo || samePhone || sameEmail;
```
Comment (gohqingkhang) [inline src/main/java/seedu/address/model/person/Person.java:206]:
```suggestion
        return (sameClass && sameRegNo) || samePhone || sameEmail;
```
not too sure if returns the same result, but just to be sure

**Review by gohqingkhang — COMMENTED** [review]
LGTM, but someone else to check also, as need to be more careful about codebase change

File: src/main/java/seedu/address/model/person/Person.java (hunk 1 above)
Comment (Shruthiiii03) [reply src/main/java/seedu/address/model/person/Person.java:206]: okay yup makes sense to be safer

File: src/main/java/seedu/address/model/person/Person.java [hunk 2]
```diff
@@ -190,16 +190,20 @@ public HashMap<AbsentDate, AbsentReason> getAttendances() {
...
     public boolean isSamePerson(Person otherPerson) {
-        if (otherPerson == this) {
-            return true;
+        if (otherPerson == null) {
+            return false;
         }
 
-        return otherPerson != null
-                && otherPerson.getName().equals(getName());
+        boolean sameClass = (otherPerson.getStudentClass().equals(this.getStudentClass()));
```
Comment (BunnyHoppp) [inline src/main/java/seedu/address/model/person/Person.java:201]: Perhaps you can make the boolean isSameClass to sound more like a boolean

File: src/main/java/seedu/address/model/person/Person.java [hunk 3]
```diff
@@ -190,16 +190,20 @@ public HashMap<AbsentDate, AbsentReason> getAttendances() {
...
+        boolean sameRegNo = (otherPerson.getRegisterNumber().equals(this.getRegisterNumber()));
```
Comment (BunnyHoppp) [inline src/main/java/seedu/address/model/person/Person.java:202]: same for the rest

File: src/main/java/seedu/address/model/person/Person.java (hunk 2 above)
Comment (Shruthiiii03) [reply src/main/java/seedu/address/model/person/Person.java]: Yup sounds good!

**Pull request merged**